### DISCUSSION TITLE
devbuild fails @ HEAD

### DISCUSSION THREAD

**Discussion by jchoover** [wixtoolset/issues#8174] [WiX Development]
Trying to get back in to Wix and fix the cert issue that is assigned to me, but I am unable to get the default branch to build. Initial issue was reported (doc bug - issues/8173), but now devbuild is failing on the unit tests.

![image](https://github.com/wixtoolset/issues/assets/2729285/5bb27cfd-a9e7-4e5d-b12c-7a7e13da86e3)

Is there a means of getting a MSBuild binary log w/ devbuild?

**Comment by barnson** [comment]
You'd have to add `-bl` to the right .cmd file. The test failure says something's busted. Did you switch to the new `main` branch?

**Reply by jchoover** [reply]
Yes.

C:\Temp\wix4>git remote --verbose
origin  git@github.com:jchoover/wix4.git (fetch)
origin  git@github.com:jchoover/wix4.git (push)
upstream        https://github.com/wixtoolset/wix.git (fetch)
upstream        https://github.com/wixtoolset/wix.git (push)

C:\Temp\wix4>git status
On branch main
Your branch is up to date with 'upstream/main'.

nothing to commit, working tree clean

**Reply by jchoover** [reply]
Would the wix.trx assist in diagnosing?  And are there any security concerns w/ publicly posting it, or should I just email it to you Bob?

**Reply by barnson** [reply]
The `WiX v4 is in effect!` means versioning is messed up, so the tests will pass when that's fixed. Do you have these three commits at the top?

```
commit f9569df0316d0ee4b0564eb900154cdca50a75f0 (HEAD -> main, origin/main)
Author: Bob Arnson <bob@firegiant.com>
Date:   Sat Apr 27 23:37:43 2024 -0400

commit ca8a55ed44bd739e5eb7d77fbcdb3eba3ba0fee8
Author: Rob Mensching <rob@firegiant.com>
Date:   Thu Apr 25 11:12:22 2024 -0700

commit a500693e3c9e94304c8b9a8229203b2b750900aa
Author: Rob Mensching <rob@firegiant.com>
Date:   Thu Apr 25 11:10:50 2024 -0700
```

**Reply by jchoover** [reply]
C:\Temp\wix4>git log
commit f9569df0316d0ee4b0564eb900154cdca50a75f0 (HEAD -> main, upstream/main, origin/master, origin/develop, origin/HEAD, develop)
Author: Bob Arnson <bob@firegiant.com>
Date:   Sat Apr 27 23:37:43 2024 -0400

    Add UTF-8, no-prefix, and Unicode characters.

commit ca8a55ed44bd739e5eb7d77fbcdb3eba3ba0fee8
Author: Rob Mensching <rob@firegiant.com>
Date:   Thu Apr 25 11:12:22 2024 -0700

    Update to wixext folder to v6

commit a500693e3c9e94304c8b9a8229203b2b750900aa
Author: Rob Mensching <rob@firegiant.com>
Date:   Thu Apr 25 11:10:50 2024 -0700

    Update tests to use TestVersion property instead of generic Version property

    The MSBuild Version property is well-known and can be set by standard
    processes. That can cause problem in the tests where the version is explicitly
    set to validate WiX behaviors. So, we've switched to using the TestVersion
    property that will be specific for our testing purposes.

commit 4983aeeb2da6e96bc8aad698752b3b962f58e585
Author: Rob Mensching <rob@firegiant.com>
Date:   Sun Apr 7 03:26:12 2024 -0700

    Move to ReleaseFlow and SomeVer

commit 973a1764223d7313beef03a66f6218905b07870b
Author: Rob Mensching <rob@firegiant.com>
Date:   Sun Apr 7 02:49:54 2024 -0700

    Start WiX Toolset v6

commit 41e11442b2ca93e444b60213b5ae99dcbab787d8 (tag: v5.0.0)
Author: Rob Mensching <rob@firegiant.com>
Date:   Fri Apr 5 09:51:01 2024 -0700

    WiX Toolset v5.0.0

commit 681cf4a9eb6be7e4092c6e5b690773fbd8469e63
Author: Rob Mensching <rob@firegiant.com>
Date:   Thu Apr 4 15:24:34 2024 -0700

    Ensure elevated SFXCA uses Windows Installer cache and unelevated uses Temp folder

    Fixes 8078

commit 6ed045ee1fe69be037a999df2e57122a25f0dedf
Author: Rob Mensching <rob@firegiant.com>
Date:   Wed Mar 27 12:15:44 2024 -0700

    Add a managed custom action E2E test

**Reply by barnson** [reply]
I'd add a `-bl` to src\build_init.cmd. That's where versions are set.

**Reply by jchoover** [reply]
That v4 error is gone (didn't change anything, just a fresh Dev Tools console and a dev build.  Now I am failing at:

  Passed WixToolsetTest.CoreIntegration.NakedFileFixture.CanBuildNakedFilesInFragments [181 ms]
  Passed WixToolsetTest.CoreIntegration.HarvestFilesFixture.CanHarvestFilesInDirectory [182 ms]
[xUnit.net 00:00:03.51]     WixToolsetTest.CoreIntegration.ExtensionFixture.TryingToAddAV4ExtensionFails [FAIL]
[xUnit.net 00:00:03.52]       System.InvalidOperationException : Sequence contains no elements
[xUnit.net 00:00:03.52]       Stack Trace:
[xUnit.net 00:00:03.52]            at System.Linq.ThrowHelper.ThrowNoElementsException()
[xUnit.net 00:00:03.52]            at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
[xUnit.net 00:00:03.52]         C:\Temp\wix4\src\wix\test\WixToolsetTest.CoreIntegration\ExtensionFixture.cs(364,0): at WixToolsetTest.CoreIntegration.ExtensionFixture.TryingToAddAV4ExtensionFails()
[xUnit.net 00:00:03.52]     WixToolsetTest.CoreIntegration.ExtensionFixture.CanManipulateExtensionCache [SKIP]
[xUnit.net 00:00:03.52]       Depends on a v5 extension being available, which isn't true for nuget.org yet or this early in the build.
  Passed WixToolsetTest.CoreIntegration.PreprocessorFixture.VariableRedefinitionIsAWarning [63 ms]
  Passed WixToolsetTest.CoreIntegration.PreprocessorFixture.SupportParensInEnvironmentVariables [< 1 ms]
  Failed WixToolsetTest.CoreIntegration.ExtensionFixture.TryingToAddAV4ExtensionFails [195 ms]
  Error Message:
   System.InvalidOperationException : Sequence contains no elements
  Stack Trace:
     at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.Single[TSource](IEnumerable``1 source)
   at WixToolsetTest.CoreIntegration.ExtensionFixture.TryingToAddAV4ExtensionFails() in C:\Temp\wix4\src\wix\test\WixToolsetTest.CoreIntegration\ExtensionFixture.cs:line 364
  Skipped WixToolsetTest.CoreIntegration.ExtensionFixture.CanManipulateExtensionCache [1 ms]
  Error Message:
   Depends on a v5 extension being available, which isn't true for nuget.org yet or this early in the build.

  Passed WixToolsetTest.CoreIntegration.MsiPackageFixture.CanDefaultPermanentMsiPackageToVisible [443 ms]

**Comment by jchoover** [comment]
What sucks is:
:: Test
dotnet test ^
 %_B%\test\WixToolsetTest.Converters\net6.0\WixToolsetTest.Converters.dll ^
 %_B%\test\WixToolsetTest.Converters.Symbolizer\net472\WixToolsetTest.Converters.Symbolizer.dll ^
 %_B%\test\WixToolsetTest.Core\net6.0\WixToolsetTest.Core.dll ^
 %_B%\test\WixToolsetTest.Core.Native\net6.0\win-x64\WixToolsetTest.Core.Native.dll ^
 %_B%\test\WixToolsetTest.CoreIntegration\net6.0\WixToolsetTest.CoreIntegration.dll ^
 %_B%\test\WixToolsetTest.BuildTasks\net472\WixToolsetTest.BuildTasks.dll ^
 %_B%\test\WixToolsetTest.Sdk\net472\WixToolsetTest.Sdk.dll ^
 --nologo -l "trx;LogFileName=%_L%\TestResults\wix.trx" || exit /b

from wix.cmd is failing, but it isn't producing the wix.trx.  I did add --logger "console;verbosity=detailed" to get more info, but not sure where I should be going with this.

**Reply by barnson** [reply]
Try running the test from VS so you can debug the test.

**Reply by jchoover** [reply]
Exit code is 2, and messages is empty...

![image](https://github.com/wixtoolset/issues/assets/2729285/86a8e63b-8ee0-4c77-b44a-7091f7ca00e0)

**Reply by barnson** [reply]
Looks like that could happen if there was a failure getting to nuget.org. Try running `wix extension add WixToolset.UI.wixext/4.0.4` to see if you get visible error messages.

**Reply by jchoover** [reply]
wix from where?

C:\Temp\wix4>build\wix\debug\net472\wix extension add WixToolset.UI.wixext/4.0.4
WixToolset.UI.wixext : warning WIX6101: Could not find expected package root folder wixext6. Ensure WixToolset.UI.wixext/4.0.4 is compatible with WiX v6.

**Reply by barnson** [reply]
What?! That makes no sense. Sorry, out of ideas. I'd start stepping through ExtensionCacheManager to see why the command line shows the warning and the test doesn't.

**Reply by jchoover** [reply]
When attempting to debug in VS:
Seems like it's only looking in the NuGet VS source via:  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

CLI build shows:

C:\Temp\wix4>dotnet nuget list source
Registered Sources:
  1.  nuget [Enabled]
      https://api.nuget.org/v3/index.json
  2.  build [Enabled]
      C:\Temp\wix4\build\artifacts

**Comment by jchoover** [comment]
notepad C:\Users\\%USERNAME%\AppData\Roaming\NuGet\NuGet.Config

Having:
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration/>
```
= Fail.

Copying from notepad C:\Temp\wix4\NuGet.Config, and deleting the build source has the build grinding away happily now.  Does vscode/python ever do anything to touch the per user nuget config?

**Reply by barnson** [reply]
I've heard of it but never seen it on one of my machines. It's complicated by the fact that VS doesn't ship nuget.exe.

**Comment by jchoover** [comment]
Umm...

C:\Temp\wix4>nuget locals all -list
http-cache: C:\Users\\%USERNAME%\AppData\Local\NuGet\v3-cache
global-packages: C:\Users\\%USERNAME%\\.nuget\packages\
temp: C:\Users\\%USERNAME%\AppData\Local\Temp\NuGetScratch
plugins-cache: C:\Users\\%USERNAME%\AppData\Local\NuGet\plugins-cache

C:\Temp\wix4>dotnet nuget list source
Registered Sources:
  1.  nuget [Enabled]
      https://api.nuget.org/v3/index.json
  2.  build [Enabled]
      C:\Temp\wix4\build\artifacts

C:\Temp\wix4>nuget help
NuGet Version: 6.9.1.3
usage: NuGet <command> [args] [options]
Type 'NuGet help <command>' for help on a specific command.

Available commands:...

**Comment by jchoover** [comment]
WTB https://github.com/NuGet/Home/issues/8420 because something isn't right between nuget on the path and dotnet test.
